### PR TITLE
Document `.clone()`

### PR DESCRIPTION
**Pull request by dns2utf8**
The example became rather long but it is the shortest one I could come up with with one pool.
I am open to suggestions.

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -464,6 +463,78 @@ impl ThreadPool {
...
 }
 
+impl Clone for ThreadPool {
+    /// Cloning a pool will create a new handle to the pool.
+    /// The behavior is similar to [Arc](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html).
+    ///
+    /// We could for example [`join`](#method.join) a pool from multiple
+    /// threads or submit jobs from multiple threads concurrently.
+    ///
+    /// ```
```
Comment (frewsxcv) [inline lib.rs]: Sorry for the delay, I'm thinking if there's a simpler example we can put here. I know you're spawning threads so you can show that the `ThreadPool` can be cloned, but do we need much of the other code here?

Comment (dns2utf8) [inline lib.rs]: No worries 😄 
I could shorten the code by half by using a loop to spawn the threads and then use another loop/fold to join them. Or would you go into a completely different direction?

**Comment by dns2utf8** [comment]
I shortened the example. Reduced the example to submitting jobs from multiple threads.

**Comment by frewsxcv** [comment]
thanks!

bors r+

**Pull request merged**